packages/compass-crud/src/components/insert-json-document.tsx

```diff
@@ -23,34 +34,86 @@ type InsertJsonDocumentProps = {
   updateJsonDoc: (value: string) => void;
 };
 
+function blankLeafValues(jsonDoc: string) {
+  return (
+    jsonDoc
+      // strings
+      .replace(/: "[^"]+"/g, ': ""')
+      // numbers
+      .replace(/: -?\d+(\.\d+)?/g, ': ')
+      // booleans
+      .replace(/: (true|false)/g, ': ')
```

Comment:
Are we using regular expressions here because we want this to work when the input isn’t valid JSON? It’s surprising, given that we already start out with valid JSON

Reply:
I just hacked it because I agree this is weird UX.

But how else would you turn { foo: 5 } into { foo: }? I guess we can walk the object and serialise it ourselves? Would also require valid JSON and in all cases we seem to have valid JSON formatted at this point anyway.

Reply:
We could walk the object, replace values that we want to get rid of with a placeholder (that is sufficiently unlikely to appear in real-world usage) and then do string replacement to substitute it with an empty string. I think that would be a bit cleaner than regexes, but yeah, it does require us starting off with something that’s reliably parsable.